File: go/0006.go
/* file: 0006.go
 * by  : agaric
 * desc: project euler #6 - "sum square difference"
 */

package main

import "fmt"

func sumsq(n int) int {
	sum := 0
	for i := 1; i <= n; i++ {
		sum += i * i
	}
	return sum
}

func sqsum(n int) int {
	sum := 0
	for i := 1; i <= n; i++ {
		sum += i
	}
	return sum * sum
}

func main() {
	n := 100
	fmt.Println(sqsum(n) - sumsq(n))
}

File: haskell/0006.hs
-- file: 0006.hs
-- by  : agaric
-- copy: public domain
-- desc: project euler #6 - "sum square difference"
-- lang: haskell

module Main (main) where

main :: IO ()
main = print answer

answer :: Integer
answer = (sum [1..100])^2 - sum (map (^2) [1..100])
